src/platform_impl/macos/monitor.rs

```diff
@@ -23,7 +27,7 @@ pub struct VideoMode {
     bit_depth: u16,
     refresh_rate_millihertz: u32,
     pub(crate) monitor: MonitorHandle,
-    pub(crate) native_mode: NativeDisplayMode,
+    pub(crate) native_mode: Arc<Mutex<NativeDisplayMode>>,
```

Comment:
This how I made this `Sync` and wrapped in in `Arc` to make it `Clone` as well.
Let me know what you think.

Reply:
I'd really like to address https://github.com/rust-windowing/winit/pull/2646, since that would give me a clearer picture of what's expected from `VideoMode`: Is it just a static blob of data? If so, then we should perhaps remove the `native_mode` field? If not, then perhaps nothing should be cached, and it'd be more clear what to do here.

In any case, I think `CGDisplayModeRef` (and that means `NativeDisplayMode`) may actually be `Send + Sync` natively, by the argument that you can access global instances (e.g. `CGDisplay::main().display_mode()`) from any thread, and that the data is immutable (at least from the view of the user).

Reply:
Don't we need some handle to live in `VideoMode` to be able to spawn the window in specific monitors and such? In any case, I'm pretty much clueless about this. Currently we have a couple of types that should be connected to the `EventLoop` but aren't, e.g. like `VideoMode` and `CustomCursor`, so they both need the same solution.

There is a connection to #3317 here, which also requires some connection to `EventLoop`, one proposal is to introduce a lifetime, which could be introduced to `VideoMode` as well.

I guess if it's possible to make this just a static blob of data, it would be by far the best solution in my opinion.

In any case, in the meantime, I will just implement `Sync` on `NativeDisplayMode` and await your confirmation.

Reply:
> need some handle to live in `VideoMode` to be able to spawn the window in specific monitors and such? 

Right. Maybe we should rename it `VideoModeHandle` to reflect that it isn't just static data?

> lifetime on `VideoMode`

Not really required for macOS, since the stuff is `'static` and able to live after `EventLoop` has been dropped.

> implement `Sync` on `NativeDisplayMode` and await your confirmation.

I'll accept that as "sound enough" for now.

Reply:
> > need some handle to live in `VideoMode` to be able to spawn the window in specific monitors and such?
> 
> Right. Maybe we should rename it `VideoModeHandle` to reflect that it isn't just static data?

I think it's a good idea! I started tracking this here: #3258.
Will actually make a PR afterwards.

> > implement `Sync` on `NativeDisplayMode` and await your confirmation.
> 
> I'll accept that as "sound enough" for now.

Would you mind giving your approval to the PR then? Just to make it official :sweat_smile:.